examples/step-67/step-67.cc

```diff
@@ -1465,60 +1505,362 @@ namespace Euler_DG
     LinearAlgebra::distributed::Vector<Number> &      solution,
     LinearAlgebra::distributed::Vector<Number> &      next_ri) const
   {
-    {
-      TimerOutput::Scope t(timer, "rk_stage - integrals L_h");
-
-      for (auto &i : inflow_boundaries)
-        i.second->set_time(current_time);
-      for (auto &i : subsonic_outflow_boundaries)
-        i.second->set_time(current_time);
-
-      data.loop(&EulerOperator::local_apply_cell,
-                &EulerOperator::local_apply_face,
-                &EulerOperator::local_apply_boundary_face,
-                this,
-                vec_ki,
-                current_ri,
-                true,
-                MatrixFree<dim, Number>::DataAccessOnFaces::values,
-                MatrixFree<dim, Number>::DataAccessOnFaces::values);
-    }
-
+    if (use_ecl)
+      {
+        TimerOutput::Scope t(timer, "rk_stage - integrals L_h");
+
+        for (auto &i : inflow_boundaries)
+          i.second->set_time(current_time);
+        for (auto &i : subsonic_outflow_boundaries)
+          i.second->set_time(current_time);
+
+        FEEvaluation<dim, degree, n_points_1d, dim + 2, Number> phi(data);
+        FEEvaluation<dim, degree, n_points_1d, dim + 2, Number> phi_temp(data);
+
+        FEFaceEvaluation<dim, degree, n_points_1d, dim + 2, Number> phi_m(data,
+                                                                          true);
+        FEFaceEvaluation<dim, degree, n_points_1d, dim + 2, Number> phi_p(
+          data, false);
+
+        Tensor<1, dim, VectorizedArray<Number>> constant_body_force;
+        const Functions::ConstantFunction<dim> *constant_function =
+          dynamic_cast<Functions::ConstantFunction<dim> *>(body_force.get());
+
+        if (constant_function)
+          constant_body_force = evaluate_function<dim, Number, dim>(
+            *constant_function, Point<dim, VectorizedArray<Number>>());
+
+        const dealii::internal::EvaluatorTensorProduct<
+          dealii::internal::EvaluatorVariant::evaluate_evenodd,
+          dim,
+          n_points_1d,
+          n_points_1d,
+          VectorizedArray<Number>>
+          eval(AlignedVector<VectorizedArray<Number>>(),
+               data.get_shape_info().data[0].shape_gradients_collocation_eo,
+               AlignedVector<VectorizedArray<Number>>());
+
+        AlignedVector<VectorizedArray<Number>> buffer(phi.static_n_q_points *
+                                                      phi.n_components);
+
+        data.template loop_cell_centric<
+          LinearAlgebra::distributed::Vector<Number>,
+          LinearAlgebra::distributed::Vector<Number>>(
+          [&](const auto &, auto &dst, const auto &src, const auto cell_range) {
+            for (unsigned int cell = cell_range.first; cell < cell_range.second;
+                 ++cell)
+              {
```

Comment:
I know you like lambdas, but for the sake of education let us move this large function into a separate named function that can be easily identified in all tools (debuggers, profilers, ...) and is consistent with the rest of the program.